src/App/App.js

```diff
@@ -92,15 +116,18 @@ class App extends Component {
                <>
                 <button className="App-nav-button" onClick={this.logOut}>{btnTxt}</button>
                 <input className="App-search-input" placeholder="Search Movies..."></input><button className="App-search-button"></button>
-                <h2 className="App-welcome-user">Welcome, Greg!</h2>
+                <h2 className="App-welcome-user" >Welcome, {this.userName}!</h2>
               </>
               }
           </nav>
         </header>
 
-        {this.state.pageDisplayed === 'login' && <Login validateLogin={this.validateLogin} action={this.logIn}/>}
-        {this.state.pageDisplayed === 'home' && <MovieContainer movies={this.state.movies} showMovieDetails={this.showMovieDetails}/>}
-        {this.state.pageDisplayed === 'moviePage' && <MoviePage foundMovieId={this.state.foundMovieId.id}/>}
+        {this.state.pageDisplayed === 'login' && 
+          <Login validateLogin={this.validateLogin} action={this.logIn} userId={this.userId}/>}
+        {this.state.pageDisplayed === 'home' && 
+          <MovieContainer movies={this.state.movies} showMovieDetails={this.showMovieDetails} isLoggedIn={this.state.isLoggedIn} userRatings={this.state.userRatings}/>}
+        {this.state.pageDisplayed === 'moviePage' && 
+          <MoviePage foundMovieId={this.state.foundMovieId.id} userRating={this.state.foundMovieRating}/>}
```

Comment:
Great candidate for a React Router implementation! Try and take a crack at implementing it tonight & do some pre-teaching before tomorrow's lesson! The lesson plan will be a great resource for this.